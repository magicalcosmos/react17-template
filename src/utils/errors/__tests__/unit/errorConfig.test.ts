import ErrorFile from '~@/utils/errors/errorConfig';

test('errorConfig 配置测试',()=>{
    expect(ErrorFile).toMatchSnapshot();
})