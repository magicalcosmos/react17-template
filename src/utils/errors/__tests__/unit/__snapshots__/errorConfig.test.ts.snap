// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorConfig 配置测试 1`] = `
Object {
  "errorCode": Object {
    "ExCodeActionFail": 15000,
    "ExCodeTokenNotBeEmpty": 12002,
    "ExCode_Auth_NoLogin": 12000,
    "ExCode_Auth_NoPermission": 12003,
    "ExCode_Auth_TokenOverdue": 12001,
    "ExCode_Common_IdGenFailed": 15009,
    "ExCode_Db_Err": 11001,
    "ExCode_DemandNameNotEmpty_Err": 28007,
    "ExCode_DemandNotFound_Err": 28005,
    "ExCode_DemandRepeatName_Err": 28003,
    "ExCode_DemandSameName_Err": 27011,
    "ExCode_DemandVersionInvalidFileType_Err": 27009,
    "ExCode_DemandVersionLspeckRequired_Err": 27008,
    "ExCode_DemandVersionLspeckUpload_Err": 27001,
    "ExCode_DemandVersionNameNotEmpty_Err": 27007,
    "ExCode_DemandVersionNatureUpload_Err": 27002,
    "ExCode_DemandVersionNotAllowed_Err": 27006,
    "ExCode_DemandVersionNotFound_Err": 27005,
    "ExCode_DemandVersionPspeckUpload_Err": 27010,
    "ExCode_DemandVersionRepeatName_Err": 27003,
    "ExCode_DemandVersion_NotFoundSafeRequirement": 27602,
    "ExCode_DownloadNotFound_Err": 26001,
    "ExCode_File_CopyFail": 15013,
    "ExCode_File_DeleteFail": 15015,
    "ExCode_File_Exist": 15018,
    "ExCode_File_MakeDirFail": 15014,
    "ExCode_File_MoveFail": 15012,
    "ExCode_File_Not_Exist": 15019,
    "ExCode_File_OpenFail": 15010,
    "ExCode_File_ReadFail": 15011,
    "ExCode_History_Err": 25000,
    "ExCode_License_Err": 21000,
    "ExCode_License_OverUserLimit": 21002,
    "ExCode_License_Overdue": 21001,
    "ExCode_License_UpdateSuccess": 21005,
    "ExCode_License_UploadFailed": 21003,
    "ExCode_License_UserLimited": 21006,
    "ExCode_Mq_Err": 11003,
    "ExCode_Online_Err": 24000,
    "ExCode_Online_UserConnByNew": 24001,
    "ExCode_Param_BadParam": 13009,
    "ExCode_Param_EmptyID": 13007,
    "ExCode_Param_EmptyName": 13008,
    "ExCode_Param_EmptyParam": 13001,
    "ExCode_Param_FaultFileSize": 13006,
    "ExCode_Param_FaultFileType": 13005,
    "ExCode_Param_FaultParamFormat": 13002,
    "ExCode_Param_FaultParamLength": 13004,
    "ExCode_Param_FaultParamType": 13003,
    "ExCode_Param_NoParam": 13000,
    "ExCode_ProjectParser_Locked": 23601,
    "ExCode_Project_CancelFailed": 23401,
    "ExCode_Project_FileName_Err": 23103,
    "ExCode_Project_FileTypeID_Err": 23102,
    "ExCode_Project_FileUploadFailed": 23101,
    "ExCode_Project_FindFailed": 23001,
    "ExCode_Project_Locked": 23402,
    "ExCode_Project_ValidationFailed": 23500,
    "ExCode_Project_ValidationLocked": 23501,
    "ExCode_Record_ExistsID": 14001,
    "ExCode_Record_RepeatName": 14002,
    "ExCode_Redis_Err": 11002,
    "ExCode_SSO_Err": 11004,
    "ExCode_Server_Err": 11000,
    "ExCode_Validation_LackOfDemandVersion": 23512,
    "ExCode_Validation_LackOfDemandVersionAndFile": 23511,
    "ExCode_Validation_LackOfFile": 23513,
    "NoPermission": 32401,
    "NoPermissionCreateUser": 32402,
    "ParamNotBeEmpty": 32405,
    "ProjectConfirmFailed": 23400,
    "ProjectDeleteFailed": 23300,
    "ProjectFindFailed": 23001,
    "ProjectHasExist": 23102,
    "SUCCESS": 200,
    "UserDeleteFailed": 32300,
    "UserHasExist": 32102,
    "UserHasNoFound": 32005,
    "UserModifyFailed": 32200,
    "UserNameTooLong": 32404,
    "UserNoFound": 32001,
    "UserPwdModifyFailed": 32202,
    "UserRoleNoFound": 32002,
    "schemaFail": 400,
    "serverFail": 500,
  },
  "errorTip": Object {
    "11000": "当前服务无法使用",
    "11001": "数据库服务出错",
    "11002": "Redis服务出错",
    "11003": "消息队列服务出错",
    "11004": "SSO服务调用出错",
    "12000": "未登录授权",
    "12001": "Token已失效",
    "12002": "Token不能为空",
    "12003": "角色权限不允许",
    "13000": "参数缺失",
    "13001": "参数不能为空",
    "13002": "参数格式有误",
    "13003": "参数类型不正确",
    "13004": "参数长度不正确",
    "13005": "文件类型不正确",
    "13006": "文件大小不正确",
    "13007": "ID不能为空",
    "13008": "名称不能为空",
    "13009": "参数格式有误",
    "14001": "ID记录已存在",
    "14002": "名称重复",
    "15000": "当前操作失败",
    "15009": "ID生成失败",
    "15010": "文件打开失败",
    "15011": "文件读取失败",
    "15012": "文件移动失败",
    "15013": "文件复制失败",
    "15014": "创建目录失败",
    "15015": "文件删除失败",
    "15018": "该文件名称已存在",
    "15019": "该文件不存在",
    "200": "成功",
    "21000": "许可证失效",
    "21001": "许可证已过期失效，请联系供应商获取新的许可证",
    "21002": "系统监测到已有用户且超出了许可证的用户上限",
    "21003": "许可证上传失败",
    "21005": "许可证被更新，请重新登录",
    "21006": "已超出在线人数范围",
    "23001": "项目不存在",
    "23101": "上传项目文件失败",
    "23102": "项目名称已经存在",
    "23103": "文件名错误",
    "23300": "删除项目错误",
    "23400": "项目保存配置失败",
    "23401": "项目取消上传失败",
    "23402": "项目被锁定",
    "23500": "apiError.ExCode_Project_ValidationFailed",
    "23501": "暂无法解析，与当前需求基线关联的项目正在运行中，请稍后再试。",
    "23511": "apiError.ExCode_Validation_LackOfDemandVersionAndFile",
    "23512": "apiError.ExCode_Validation_LackOfDemandVersion",
    "23513": "apiError.ExCode_Validation_LackOfFile",
    "23601": "项目解析锁定",
    "24000": "在线状态异常",
    "24001": "有信连接打开",
    "25000": "查询历史记录异常",
    "26001": "没有找到需要下载的数据",
    "27001": "Lspeck语言文件上传错误",
    "27002": "自然语言文件上传错误",
    "27003": "需求基线名称重名",
    "27005": "需求基线版本不存在",
    "27006": "需求基线不允许删除",
    "27007": "需求基线名称不能为空",
    "27008": "Lspeck语言文件必传",
    "27009": "文件类型不正确",
    "27010": "Pspec语言文件上传错误",
    "27011": "需求基线名称重名",
    "27602": "没有找到安全需求",
    "28003": "需求基线名称重名",
    "28005": "需求基线不存在",
    "28007": "需求基线名称不能为空",
    "32001": "用户未找到",
    "32002": "没找到合适记录",
    "32005": "用户未找到",
    "32102": "用户已存在",
    "32200": "修改用户信息失败",
    "32202": "修改密码错误",
    "32300": "用户删除失败",
    "32401": "没有权限操作",
    "32402": "用户没有权限创建用户",
    "32404": "用户名过长",
    "32405": "必填参数不能为空",
    "400": "服务端接收参数错误",
    "500": "服务端错误",
  },
  "needTipCode": Array [
    400,
    500,
    11000,
    11001,
    11002,
    11003,
    11004,
    12000,
    12001,
    12003,
    13000,
    13001,
    13002,
    13003,
    13004,
    13005,
    13006,
    13007,
    13008,
    13009,
    14001,
    14002,
    15000,
    15009,
    15010,
    15011,
    15012,
    15013,
    15014,
    15015,
    15018,
  ],
}
`;
